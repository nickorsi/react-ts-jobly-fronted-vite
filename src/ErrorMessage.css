.error-container {
  margin: auto 15px;
}

.error-container h2 {
  text-align: center;
  color: white;
  text-shadow: 2px 2px 5px rgb(52, 48, 47);
  font-weight: 700;
  font-size: 3rem;
}

.error-container p {
  color: white;
  text-align: center;
  font-size: 2rem;
}