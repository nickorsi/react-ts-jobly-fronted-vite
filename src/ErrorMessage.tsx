import "./ErrorMessage.css";

/**
 * ErrorMessage component that renders a graceful error message to the user.
 *
 * Take props {errors}, with errors being an array of string error messages.
 *
 * {CompanyList, JobList, Company} -> ErrorMessage
 */
function ErrorMessage({ errors }: {'errors': string[]}) {
    return (
        <div className="error-container">
            <h2>Oops!</h2>
            <p>Looks like something went wrong:
            {errors.map((er, i, arr) => {
                return i + 1 !== arr.length ?
                    (<span key={i}> {er},</span>) :
                    (<span key={i}> {er}</span>)
            })}
            </p>
        </div>
    )
};

export default ErrorMessage;