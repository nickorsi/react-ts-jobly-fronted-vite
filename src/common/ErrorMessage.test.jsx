import {describe, expect, it} from "vitest";
import { render } from "@testing-library/react";

import ErrorMessage from "./ErrorMessage.tsx";

describe("Simple Render Tests", () => {
    it("renders without crashing", function () {
        render(<ErrorMessage errors={["Testing this!"]}/>);
    })
    it("matches the snapshot", function () {
        const { container } = render(
            <ErrorMessage errors={["Testing this!"]}/>
        );

        expect(container).toMatchSnapshot();
    })
})

describe("Multiple Errors Tests", () => {
    it("renders the multiple errors correctly", function () {
        const { container } = render(
            <ErrorMessage
                errors={["Did", "you", "get", "all", "this"]}
            />
        )

        expect(
            container.querySelector(".error-container p").textContent
        ).toEqual(
            "Looks like something went wrong: Did, you, get, all, this"
        )
    })
})