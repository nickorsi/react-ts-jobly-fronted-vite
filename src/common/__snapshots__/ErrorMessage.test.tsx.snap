// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Render Tests > matches the snapshot 1`] = `
<div>
  <div
    class="error-container"
  >
    <h2>
      Oops!
    </h2>
    <p>
      Looks like something went wrong:
      <span>
         
        Testing this!
      </span>
    </p>
  </div>
</div>
`;
